### ISSUE TITLE
`docker run --rm` incompatible with `restart: always`

### ISSUE THREAD

**Issue by wernight** [docker/compose#1218]
Having a `docker-compose.yml` with `restart: always` is good for services started by `docker up`. However when executing commands during a build or during development via `docker run --rm ...` it'll also restart that container and this command will loop forever.

* Could allow specifying `--restart never` for `docker run` but that's edge.
* Should ignore `restart: ` during `docker-compose run ...` commands, only use it for `docker-compose start` or `docker-compose up`
* Could restart linked containers even if they don't have `restart: always`. This would also fix the issue as one could do:

````
service:
  links:
    - web
  restart: always
web:
  image: nginx
  links: php
php:
  image: php
````

Then a `docker-compose run --rm php ...` would not restart this command, but a `docker-compose up service` would start and restart all linked services.

**Comment by wernight** [comment]
It also affects linked containers. For example:

````
web:
  image: nginx
test:
  links: php
````

    $ docker-compose run --rm test

Will start but not stop `web`.

**Comment by dnephin** [comment]
Duplicate of #1013 I think

**Comment by wernight** [comment]
Yes looks like duplicate. Should close or leave open until the other is fixed so people can find it faster?

**Comment by Neppord** [comment]
1+

the current behavior is bad and not that intuitive.

**Comment by wernight** [comment]
I think it has been fixed in 1.3.

**Comment by wernight** [comment]
`docker-compose run` without `--rm` should it respect `restart: always` or
not? I guess you're arguing it should not. I'm finding both solutions valid
and I'm more thinking about issues. If there is a flag `--restart=always`
for `docker-compose run` then I'd agree. Without that, there are cases
where it would else remove features.

**Comment by dnephin** [comment]
this was fixed in #1205
